components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/throttling/ThrottleHandler.java

```diff
@@ -229,127 +230,131 @@ private boolean doRoleBasedAccessThrottlingWithCEP(MessageContext synCtx, Config
+                                context1 = timer1.start();
+
+                        if (conditionGroupDTOs != null && conditionGroupDTOs.length > 0) {
+                            // Checking Applicability of Conditions is a relatively expensive operation. So we are
+                            // going to check it only if the API/Resource is throttled out.
+                            if (getThrottleDataHolder().isAPIThrottled
+                                    (resourceLevelThrottleKey)) {
+                                if (getThrottleDataHolder().isConditionsAvailable(resourceLevelThrottleKey)) {
+                                    Map<String, List<ConditionDto>> conditionDtoMap = getThrottleDataHolder()
+                                            .getConditionDtoMap(resourceLevelThrottleKey);
                                     if (log.isDebugEnabled()) {
-                                        log.debug("Throttled with Condition :" + throttledCondition);
+                                        log.debug("conditions available" + conditionDtoMap.size());
                                     }
-                                    String combinedResourceLevelThrottleKey = resourceLevelThrottleKey + "_" +
-                                            throttledCondition;
+                                    String throttledCondition = getThrottleConditionEvaluator().getThrottledInCondition
+                                            (synCtx, authContext, conditionDtoMap);
+                                    if (StringUtils.isNotEmpty(throttledCondition)) {
+                                        if (log.isDebugEnabled()) {
+                                            log.debug("Throttled with Condition :" + throttledCondition);
+                                        }
+                                        String combinedResourceLevelThrottleKey = resourceLevelThrottleKey + "_" +
+                                                throttledCondition;
 
-                                    if (log.isDebugEnabled()) {
-                                        log.debug("Checking condition : " + combinedResourceLevelThrottleKey);
-                                    }
+                                        if (log.isDebugEnabled()) {
+                                            log.debug("Checking condition : " + combinedResourceLevelThrottleKey);
+                                        }
 
-                                    if (getThrottleDataHolder().
-                                            isThrottled(combinedResourceLevelThrottleKey)) {
-                                        if (!apiLevelThrottledTriggered) {
-                                            isResourceLevelThrottled = isThrottled = true;
-                                        } else {
-                                            isApiLevelThrottled = isThrottled = true;
+                                        if (getThrottleDataHolder().
```

Comment:
does this line if combined with the line below exceed 120 characters?

Reply:
Fixed